### PR TITLE
[exporter/googlecloud] Add gRPC related metrics

### PR DESCRIPTION
**Pull request by sophieyfang**
**Description:**
This PR is for fixing a missing metrics downstream by our ops-agent: GoogleCloudPlatform/ops-agent#305. This will add grpc related metrics into googlecloud exporter.

**Link to tracking Issue:**  GoogleCloudPlatform/ops-agent#305

**Testing:** 
I vendored this googlecloud module in our agent code, then modified the files with changes in this PR, then built and ran the agent to see grpc metrics were populating as expected

```
sophieyfang_google_com@debian10-meow:~$ curl localhost:8888/metrics  | grep "grpc"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 19659    0 19659    0     0  2742k      0 --:--:-- --:--:-- --:--:-- 2742k
# HELP otelcol_grpc_io_client_completed_rpcs Count of RPCs by method and status.
# TYPE otelcol_grpc_io_client_completed_rpcs counter
otelcol_grpc_io_client_completed_rpcs{grpc_client_method="google.monitoring.v3.MetricService/CreateTimeSeries",grpc_client_status="OK",service_version="latest"} 913
# HELP otelcol_grpc_io_client_received_bytes_per_rpc Distribution of bytes received per RPC, by method.
# TYPE otelcol_grpc_io_client_received_bytes_per_rpc histogram
otelcol_grpc_io_client_received_bytes_per_rpc_bucket{grpc_client_method="google.monitoring.v3.MetricService/CreateTimeSeries",service_version="latest",le="1024"} 913
otelcol_grpc_io_client_received_bytes_per_rpc_bucket{grpc_client_method="google.monitoring.v3.MetricService/CreateTimeSeries",service_version="latest",le="2048"} 913
otelcol_grpc_io_client_received_bytes_per_rpc_bucket{grpc_client_method="google.monitoring.v3.MetricService/CreateTimeSeries",service_version="latest",le="4096"} 913
otelcol_grpc_io_client_received_bytes_per_rpc_bucket{grpc_client_method="google.monitoring.v3.MetricService/CreateTimeSeries",service_version="latest",le="16384"} 913
otelcol_grpc_io_client_received_bytes_per_rpc_bucket{grpc_client_method="google.monitoring.v3.MetricService/CreateTimeSeries",service_version="latest",le="65536"} 913
otelcol_grpc_io_client_received_bytes_per_rpc_bucket{grpc_client_method="google.monitoring.v3.MetricService/CreateTimeSeries",service_version="latest",le="262144"} 913
otelcol_grpc_io_client_received_bytes_per_rpc_bucket{grpc_client_method="google.monitoring.v3.MetricService/CreateTimeSeries",service_version="latest",le="1.048576e+06"} 913
otelcol_grpc_io_client_received_bytes_per_rpc_bucket{grpc_client_method="google.monitoring.v3.MetricService/CreateTimeSeries",service_version="latest",le="4.194304e+06"} 913
otelcol_grpc_io_client_received_bytes_per_rpc_bucket{grpc_client_method="google.monitoring.v3.MetricService/CreateTimeSeries",service_version="latest",le="1.6777216e+07"} 913
otelcol_grpc_io_client_received_bytes_per_rpc_bucket{grpc_client_method="google.monitoring.v3.MetricService/CreateTimeSeries",service_version="latest",le="6.7108864e+07"} 913
otelcol_grpc_io_client_received_bytes_per_rpc_bucket{grpc_client_method="google.monitoring.v3.MetricService/CreateTimeSeries",service_version="latest",le="2.68435456e+08"} 913
.
.
.
```

### CODE REVIEW

File: exporter/googlecloudexporter/factory.go
Comment (jpkrohling) [inline exporter/googlecloudexporter/factory.go:44]: The indentation here looks odd.

File: exporter/googlecloudexporter/factory.go
Comment (MovieStoreGuy) [inline exporter/googlecloudexporter/factory.go:45]: Two things:

- Please use zap.Logger instead of the std log implementation.
- Registering the view's here is problematic since it would break on application start up if there was an issue.

Can I ask if this call to Register be done elsewhere where the error can be handled correctly?

File: exporter/googlecloudexporter/googlecloud.go
Comment (MovieStoreGuy) [inline exporter/googlecloudexporter/googlecloud.go:79]: For my own understanding, can I ask why we haven't used an otel stats handler?

**Review by MovieStoreGuy — CHANGES_REQUESTED** [review]
I believe this breaks the expectations of a collector components of not panicing / exiting on error.

File: exporter/googlecloudexporter/factory.go
Comment (sophieyfang) [reply exporter/googlecloudexporter/factory.go:45]: It can't be done elsewhere... 
Initially, i put the view.Register(ocgrpcx...) inside googecloud.go.By doing that, the agent didn't generate any grpc metrics. After going thru how other exporters deal with agent exporter metrics, i found out they always put the view.Register inside the component.ExporterFactory, only by doing this, i tested it started to generate grpc metrics. Sample on how other exporter handle such cases https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/newrelicexporter/factory.go#L33

I can modify it to 
```
	once.Do(func() {
		view.Register(viewPointCount)
                view.Register(ocgrpc.DefaultClientViews...)
```

File: exporter/googlecloudexporter/googlecloud.go
Comment (sophieyfang) [reply exporter/googlecloudexporter/googlecloud.go:79]: This PR is for fixing a missing metrics downstream by our ops-agent: https://github.com/GoogleCloudPlatform/ops-agent/issues/305. This is not a permanent solution, more like a quick walkaround to bring this GA metrics back for our customer to use. 
In the meanwhile, @jsuereth and @punya 's team are working on rewriting this googlecloud exporter. Also side questions for not missing your sentences here: which `otel stats handler` do you mean?

File: exporter/googlecloudexporter/factory.go
Comment (MovieStoreGuy) [reply exporter/googlecloudexporter/factory.go:45]: That sounds reasonable to me, the only reason i am hesitant is that with the correct approach is it can cause panic on start up of no fault to the user and would require either downgrading the collector or trying to rollout a fix for it.

File: exporter/googlecloudexporter/googlecloud.go
Comment (MovieStoreGuy) [reply exporter/googlecloudexporter/googlecloud.go:79]: I had thought that there was a port from open census into `otel-go`, however, if that is not the case then all is good :) 

I just more wanted to understand if some functionality was missing is all.

**Review by codeboten — COMMENTED** [review]
Pinging codeowners @aabmass @dashpole @jsuereth @punya @tbarker25

File: exporter/googlecloudexporter/googlecloud.go
Comment (punya) [reply exporter/googlecloudexporter/googlecloud.go:79]: @sophieyfang is exactly correct, we are working on a reimplementation of this exporter that avoids the OpenCensus dependency. However, that will take longer to complete and test.

**Comment by punya** [comment]
@sophieyfang would you mind clarifying in the PR description that you’re adding *back* metrics that were previously present (and if possible, could you identify the commit or release in which the gRPC metrics went away)?

**Review by dashpole — APPROVED** [review]
lgtm!

File: exporter/googlecloudexporter/factory.go
Comment (sophieyfang) [reply exporter/googlecloudexporter/factory.go]: Done removing panicking :)

File: exporter/googlecloudexporter/googlecloud.go
Comment (sophieyfang) [reply exporter/googlecloudexporter/googlecloud.go:79]: > @sophieyfang would you mind clarifying in the PR description that you’re adding _back_ metrics that were previously present (and if possible, could you identify the commit or release in which the gRPC metrics went away)?

I have edit the description by linking the PR to ops-agent github issue which contains the details of how it went wrong.

**Review by codeboten — APPROVED** [review]
Thanks for submitting the fix and addressing the comments!

**Review by codeboten — COMMENTED** [review]
@dashpole can you approve again, your review was requested. @MovieStoreGuy can you confirm if your comments have been addressed.

**Comment by codeboten** [comment]
I dont believe the CI failures are related to this change

**Comment by jpkrohling** [comment]
This will be merged once @dashpole approves this change again.

**Review by dashpole — APPROVED** [review]

**Pull request merged**
